### PR TITLE
feat: make RepeatedTask invoke remove_outdated_meta method

### PR DESCRIPTION
**Pull request by NiwakaDev**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR makes a `RepeatedTask` invoke the `remove_outdated_meta` method.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

#1509

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/common/procedure/src/local.rs [hunk 1]
```diff
@@ -487,18 +517,35 @@ impl ProcedureManager for LocalManager {
...
 }
 
+struct RemoveOutdatedMetaFunction {
+    manager_ctx: Arc<ManagerContext>,
+    ttl: Duration,
+}
+
+use common_runtime::TaskFunction;
+
+use crate::error::Error;
```
Comment (evenyag) [inline src/common/procedure/src/local.rs:535]: nitpick: Could you move them to line 30 so the `cargo fmt` can merge these imports?

**Review by evenyag — APPROVED** [review]
Mostly LGTM

File: src/common/procedure/src/local.rs [hunk 2]
```diff
@@ -807,4 +858,59 @@ mod tests {
...
+        watcher.changed().await.unwrap();
+        manager.start().unwrap();
+        tokio::time::sleep(Duration::from_millis(10)).await;
+        assert!(manager
+            .procedure_state(procedure_id)
+            .await
+            .unwrap()
+            .is_none());
+
+        //The remove_outdated_meta method has been stopped, so any procedure meta-data will not be automatically removed.
```
Comment (waynexia) [inline src/common/procedure/src/local.rs:895]: style:
```suggestion
        // The remove_outdated_meta method has been stopped, so any procedure meta-data will not be automatically removed.
```

File: src/common/procedure/src/error.rs [hunk 3]
```diff
@@ -121,6 +121,18 @@ pub enum Error {
 
     #[snafu(display("Corrupted data, error: {source}"))]
     CorruptedData { source: FromUtf8Error },
+
+    #[snafu(display("Failed to start the remove_outdated_meta method"))]
```
Comment (waynexia) [inline src/common/procedure/src/error.rs:125]:
```suggestion
    #[snafu(display("Failed to start the remove_outdated_meta method, error: {source}"))]
```

File: src/common/procedure/src/error.rs [hunk 4]
```diff
@@ -121,6 +121,18 @@ pub enum Error {
...
+    StartRemoveOutdatedMetaTask {
+        source: common_runtime::error::Error,
+        location: Location,
+    },
+
+    #[snafu(display("Failed to stop the remove_outdated_meta method"))]
```
Comment (waynexia) [inline src/common/procedure/src/error.rs:131]:
```suggestion
    #[snafu(display("Failed to stop the remove_outdated_meta method, error: {source}"))]
```

**Pull request merged**